File: backend/src/main/java/live/rkozik/pm_app/backend/models/SmallTask.java
package live.rkozik.pm_app.backend.models;


import lombok.Data;
import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;


@Entity
@Table(name = "SmallTasks")
public @Data
class SmallTask {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String description;
    private boolean isDone;

    @ManyToMany
    @JoinTable(
        name = "SmallTasks_assigned",
        joinColumns = @JoinColumn(name = "id_smallTasks"),
        inverseJoinColumns = @JoinColumn(name = "id_users"))
    private List<User> projectAssigned;
    @OneToMany(targetEntity = ToDo.class, mappedBy = "smallTask")
    private List<ToDo> toDos = new ArrayList<>();
    @ManyToOne
    private HugeTask hugeTask;

}

File: frontend/src/helpers/classes/SmallTask.ts
import DataObject from "./DataObject";
import User from "./User";
import ToDo from "./ToDo";

export default class SmallTask implements DataObject {
    id:number;
    name: string;
    description: string;
    assignedUsers: Array<User>;
    toDos: Array<ToDo>;

    constructor(_id:number, _name:string, _description: string, _assignedUsers: Array<User>, _toDos: Array<ToDo>){
        this.id = _id;
        this.name = _name;
        this.description = _description;
        this.assignedUsers = _assignedUsers;
        this.toDos = _toDos;
    }
    parseFromResponse = (json:string) => {
        return null;
    };
}